lib/plugins/features/movies/controller.js

```diff
@@ -9,6 +11,20 @@ exports.create = (payload) => {
   });
 };
 
+exports.addLocation = (movie_id, payload) => {
+  return Bluebird.all([
+    new Movie({ id: movie_id }).fetch(),
+    new Location({ id: payload.id }).fetch()
```

Comment:
down below in your `.spread`, there is a possibility that `movie` and/or `location` could be `null` if the id i pass in is a bad id. if thats the case, currently your endpoint will produce a 500 (internal server error). usually our philosophy is to try to anticipate any potential errors and minimize the number of 500s that our code produces.

in this scenario, instead of 500ing, we should return a 404 saying either the movie or location is not found. this is the same behavior we have in our api if you provide a bad address if for the `to` param of the postcard endpoint.

to do this cleanly, you need to add a few things. [the `.fetch` function](http://bookshelfjs.org/#Model-instance-fetch) takes in a few options and one of them is `require: true`. this will make the promise that is returned from `.fetch` reject with [`NotFoundError`](http://bookshelfjs.org/#Model-static-NotFoundError) if it returns 0 rows from the db.

then, we can catch that error with a `.catch` and return an appropriate error with the [`boom`](https://github.com/hapijs/boom) module. it's part of the hapi library that handles all things error related.

if you need some help with this part, just let me know!